### PR TITLE
giftras

### PR DESCRIPTION
**Pull request by andyngdz**
Note. This candidate has customed the CI. To start the project uses:

`yarn start`

### CODE REVIEW

File: challenge-react/src/App.js [hunk 1]
```diff
@@ -1,111 +1,221 @@
...
       selectedAmount: 10,
     };
+    this.handlePay = this.handlePay.bind(this);
+    this.showHide = this.showHide.bind(this)
+    this.closePaymentModal = this.closePaymentModal.bind(this);
+  }
+
+  closePaymentModal() {
+    const { charities } = this.state;
+    charities.map((charity, i) => {
```
Comment (EpicHigh) [inline challenge-react/src/App.js:23]: An unused parameter `i` need to be removed.

File: challenge-react/src/App.js [hunk 2]
```diff
@@ -1,111 +1,221 @@
...
+    this.setState( {selectedAmount: 10})
+  }
+
+
+  showHide(e) {
+    const { charities } = this.state;
+    const currentCharity= e.target.id -1;
+    // To make sure other payment modal is closed before open the current one
+    this.closePaymentModal();
+    charities[currentCharity].show = !charities[currentCharity].show ? true : false;
```
Comment (EpicHigh) [inline challenge-react/src/App.js:39]: - Using an impure operator.
- This code can be simpify like this `charities[currentCharity].show = !charities[currentCharity].show`.

**Review by EpicHigh — COMMENTED** [review]
- This one is an old syntax. Currently, we are embracing a react hook API, and I prefer the new one.
- Using Promise Hell, I personally prefer async/await. Because Promise Hell makes some difficult to new developer understand what the piece of code does, this kind of syntax is an old-fashioned and Promise Hell lacks familiarity with Promises.
- Lack of code splitting, common components, and others The candidate put all of the code in a single file, which leads to spaghetti code. This is not a best practice. This kind of unstructured code is difficult-to-maintain source code. So your code must follow 5 pillars of engineering practice: maintainability, usability scalability, and reusability, extensibility.
- In Helper.js, The candidate did not edit our function and provide an initial value to our function. When the function receives an empty array, it will throw an error.
- No test cases.

File: challenge-react/src/App.js (hunk 2 above)
Comment (andyngdz) [reply challenge-react/src/App.js:39]: Good point

**Comment by andyngdz** [comment]
I have the same comment as Epic

**Comment by andyngdz** [comment]
Same comment as Tum. This candidate is 🙅

**Pull request closed without merging**